impl/src/main/java/slack/SlackResourceImpl.java

```diff
@@ -32,44 +38,103 @@ public SlackResourceImpl(SlackConfig slackConfig) {
 
     @Override
     public Observable<String> getUserEmail(String userId) {
-        return fromCallable(() -> slack.methods().usersInfo(
-            UsersInfoRequest
-            .builder()
+        return callSlack(() -> {
+
+            final UsersInfoRequest slackRequest = UsersInfoRequest
+                .builder()
                 .user(userId)
                 .token(slackConfig.getApiToken())
-            .build())
-            .getUser()
-            .getProfile()
-            .getEmail());
+                .build();
+
+            return slack.methods().usersInfo(slackRequest);
+        }).flatMap(usersInfoResponse -> {
+            if (usersInfoResponse.isOk()) {
```

Comment:
The code inside of the flatmap could be tested if the usersInfoResponse is something we can work with. If so, we could extract them to separat methods, that are package protected and test them.  The coverage will be reduced a bit otherwise or what do you think?

Reply:
Fixed